bus/api.go

```diff
@@ -107,18 +107,21 @@ type AddObjectRequest struct {
 // A Contract uniquely identifies a Sia file contract on a host, along with the
 // host's IP.
 type Contract struct {
+	ID          types.FileContractID `json:"id"`
 	HostKey     PublicKey            `json:"hostKey"`
 	HostIP      string               `json:"hostIP"`
-	ID          types.FileContractID `json:"id"`
 	StartHeight uint64               `json:"startHeight"`
 	EndHeight   uint64               `json:"endHeight"`
-	Metadata    ContractMetadata     `json:"metadata"`
+	ContractMetadata
 }
 
 // ContractMetadata contains all metadata for a contract.
 type ContractMetadata struct {
-	RenewedFrom types.FileContractID `json:"renewedFrom"`
-	Spending    ContractSpending     `json:"spending"`
+	ParentID      types.FileContractID `json:"parentID"`
+	Spending      ContractSpending     `json:"spending"`
+	GoodForUpload bool                 `json:"goodForUpload"`
+	GoodForRenew  bool                 `json:"goodForRenew"`
```

Comment:
I think we should reevaluate whether we truly need GFU/GFR, or if they're just holdovers from `siad`. That is: why do we track these things as explicit booleans, rather than computing them on-demand from other values?

Reply:
I think in `siad` we did it this way because we needed a very heavy context to determine whether a contract is good or bad and we don't always have access to all the relevant component or might not even want to have access.

e.g. to determine these fields in `siad` we needed access to the churn limiter, contractor, contract set, hostdb and probably more things that I'm missing.

Reply:
I kind of like the idea of trying to get rid of those so I went for it. If we find we need them we can also reintroduce the concept. I think we will need some form of caching here though because there are some checks that we don't want to do on the fly I think (gouging checks for instance).

Perhaps this is a good thread to talk about the contract sets as well. Currently the autopilot is managing one default set, but in my head it should manage more? E.g. `GoodForDownload` and `GoodForUpload` sets?